hibernate-core/src/main/java/org/hibernate/tool/schema/internal/SchemaCreatorImpl.java

```diff
@@ -144,13 +156,37 @@ public void doCreation(Metadata metadata, boolean createSchemas, Dialect dialect
 
 		final Set<String> exportIdentifiers = new HashSet<String>( 50 );
 
-		// first, create each schema
-		for ( Namespace namespace : database.getNamespaces() ) {
-			if ( createSchemas ) {
-				if ( namespace.getName().getSchema() == null ) {
-					continue;
+		// first, create each catalog/schema
+		if ( tryToCreateCatalogs || tryToCreateSchemas ) {
+			Set<Identifier> exportedCatalogs = new HashSet<Identifier>();
+			for ( Namespace namespace : database.getNamespaces() ) {
+
+				if ( tryToCreateCatalogs ) {
+					final Identifier catalogLogicalName = namespace.getName().getCatalog();
+					final Identifier catalogPhysicalName = namespace.getPhysicalName().getCatalog();
+
+					if ( catalogPhysicalName != null && !exportedCatalogs.contains( catalogLogicalName ) ) {
+						applySqlStrings(
+								targets, dialect.getCreateCatalogCommand(
+										catalogPhysicalName.render(
+												dialect
+										)
+								)
+						);
+						exportedCatalogs.add( catalogLogicalName );
+					}
+				}
+
+				if ( tryToCreateSchemas && namespace.getPhysicalName()
```

Comment:
Same here in terms of styling.  If you want to chop this down at all, I think what you have is less readable than:

````
if ( tryToCreateSchemas
        && namespace.getPhysicalName().getSchema() != null ) {
    ...
}
````